clippy_lints/src/lib.rs

```diff
@@ -1007,6 +1009,7 @@ pub fn register_plugins(store: &mut rustc_lint::LintStore, sess: &Session, conf:
     store.register_late_pass(|| box to_digit_is_some::ToDigitIsSome);
     let array_size_threshold = conf.array_size_threshold;
     store.register_late_pass(move || box large_stack_arrays::LargeStackArrays::new(array_size_threshold));
+    store.register_late_pass(move || box non_scalar_const::NonScalarConst::new(array_size_threshold));
```

Comment:
Uh that's an entire new situation to reuse a configuration option for multiple lints. Our documentation (generation script) can't deal with this currently.

Reply:
Hmm ok. So I need to add my proper configuration option to make it work, right?

Reply:
You can apply this patch:

```diff
diff --git a/util/lintlib.py b/util/lintlib.py
index 16cc6ccf..62fbfdca 100644
--- a/util/lintlib.py
+++ b/util/lintlib.py
@@ -14,7 +14,7 @@ lintname_re = re.compile(r'''pub\s+([A-Z_][A-Z_0-9]*)''')
 group_re = re.compile(r'''\s*([a-z_][a-z_0-9]+)''')
 conf_re = re.compile(r'''define_Conf! {\n([^}]*)\n}''', re.MULTILINE)
 confvar_re = re.compile(
-    r'''/// Lint: (\w+)\. (.*)\n\s*\([^,]+,\s+"([^"]+)":\s+([^,]+),\s+([^\.\)]+).*\),''', re.MULTILINE)
+    r'''/// Lint: ([\w,\s]+)\. (.*)\n\s*\([^,]+,\s+"([^"]+)":\s+([^,]+),\s+([^\.\)]+).*\),''', re.MULTILINE)
 comment_re = re.compile(r'''\s*/// ?(.*)''')

 lint_levels = {
@@ -93,8 +93,9 @@ def parse_configs(path):
     match = re.search(conf_re, contents)
     confvars = re.findall(confvar_re, match.group(1))

-    for (lint, doc, name, ty, default) in confvars:
-        configs[lint.lower()] = Config(name.replace("_", "-"), ty, doc, default)
+    for (lints, doc, name, ty, default) in confvars:
+        for lint in lints.split(','):
+            configs[lint.strip().lower()] = Config(name.replace("_", "-"), ty, doc, default)

     return configs

```

and then add the lint name here:

https://github.com/rust-lang/rust-clippy/blob/1ff81c1b6d7abdcc9ee47f4a8ab175082cad4421/clippy_lints/src/utils/conf.rs#L153-L154

like this:

```diff
-    /// Lint: LARGE_STACK_ARRAYS. The maximum allowed size for arrays on the stack
+    /// Lint: LARGE_STACK_ARRAYS, LARGE_CONST_ARRAYS. The maximum allowed size for arrays on the stack
```

Reply:
> So I need to add my proper configuration option to make it work, right?

With the patch above, configuration options can be reused in the future.